### PR TITLE
feat: adding shadow to tailwind preset

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This pull request adds new shadow tokens to the `design-system-tailwind-preset` package. The shadow tokens are derived from the design token CSS variables, and include both color and size definitions. These tokens will generate Tailwind utility classes that ensure consistent shadow styles across components. The shadow configuration has been imported into the index preset for global use.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-design-system/issues/57

## **Manual testing steps**

1. Run `yarn build && yarn` to build and install the local version of the tailwind preset in `storybook-react`
2. Run storybook using `yarn storybook`
3. See the shadows stories
4. Apply shadow utility classes (`shadow-default`, `shadow-primary`, `shadow-error`, etc.) to verify color.
5. Apply shadow size classes (`shadow-xs`, `shadow-sm`, `shadow-md`, `shadow-lg`) to check correct shadow sizing.
6. Verify that the shadows match the design tokens for both size and color.

### **Before**

N/A (no shadow utilities available)

### **After**

https://github.com/user-attachments/assets/14f0eb19-2a59-4234-89af-f0b8c97949b4

Testing in Portfolio using `yarn link`

https://github.com/user-attachments/assets/76b6e554-13b4-40b1-83f0-6f540c827fb2

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs)
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and/or screenshots.

### CODE REVIEW

File: packages/design-system-tailwind-preset/src/index.test.ts [hunk 1]
```diff
@@ -12,7 +13,8 @@ describe('Tailwind Preset', () => {
 
   it('configuration has correct structure', () => {
     expect(tailwindConfig).toHaveProperty('content');
-    expect(tailwindConfig).toHaveProperty('theme.extend');
+    expect(tailwindConfig).toHaveProperty('theme');
+    expect(tailwindConfig.theme).toHaveProperty('extend');
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/index.test.ts:17]: Revert these changes I think we want to ensure extend is inside of theme

File: packages/design-system-tailwind-preset/src/index.ts [hunk 2]
```diff
@@ -22,11 +23,12 @@ const tailwindConfig: Config = {
...
   },
-  plugins: [],
+  plugins: [
+    shadowPlugin, // Use the imported shadowPlugin
+  ],
 };
 
 export default tailwindConfig;
-
-module.exports = tailwindConfig;
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/index.ts:32]: Add this back for common js modules

File: apps/storybook-react/stories/Shadows.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
```
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Shadows.stories.tsx:1]: Move to a different PR and use Portfolio as the recording

File: packages/design-system-tailwind-preset/src/index.test.ts [hunk 4]
```diff
@@ -72,4 +73,18 @@ describe('Tailwind Preset', () => {
...
+
+  it('boxShadow is correctly defined in theme.extend', () => {
+    expect(tailwindConfig.theme?.extend).toHaveProperty('boxShadow');
+  });
+
+  it('boxShadow contains all expected shadow sizes', () => {
+    expect(tailwindConfig.theme?.extend?.boxShadow).toStrictEqual(
+      expect.objectContaining(shadows),
+    );
+  });
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/index.test.ts:89]: Ensures that boxShadow exists in the preset

File: apps/storybook-react/stories/Shadows.stories.tsx (hunk 3 above)
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Shadows.stories.tsx:1]: We will likely want to improve these stories and documentation in the future, but for now, we can use it as a testing sandbox. It's by no means the final product.

**Review by garrettbear — APPROVED** [review]
